### PR TITLE
feat(dlp): added a sample for inspect custom regex 

### PR DESCRIPTION
**Pull request by AarshD-crest**
- Added a sample for [inspect custom regex](https://cloud.google.com/dlp/docs/samples/dlp-inspect-custom-regex) with unit test-case.

### CODE REVIEW

File: dlp/snippets/inspect/inspect_test.go [hunk 1]
```diff
@@ -276,3 +276,21 @@ func TestInspectBigquery(t *testing.T) {
 		})
 	}
 }
+
+func TestInspectWithCustomRegex(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	buf := new(bytes.Buffer)
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_test.go:282]: issue: don't need to use the `new` keyword here. Just use `var`.

Example:
```
var buf bytes.Buffer
```

File: dlp/snippets/inspect/inspect_test.go [hunk 2]
```diff
@@ -276,3 +276,21 @@ func TestInspectBigquery(t *testing.T) {
...
+
+	if err := inspectWithCustomRegex(buf, tc.ProjectID, "Patients MRN 444-5-22222", "[1-9]{3}-[1-9]{1}-[1-9]{5}", "C_MRN"); err != nil {
+		t.Errorf("inspectWithCustomRegex: %v", err)
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_test.go:285]: issue: replace `t.Errorf()` with `t.Fatal(err)`. If the test fails here, the next test will fail, too (most likely).

File: dlp/snippets/inspect/inspect_with_custom_regex.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the type and content to be inspected.
+	var contentItem = &dlppb.ContentItem{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_with_custom_regex.go:48]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_with_custom_regex.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+		DataItem: &dlppb.ContentItem_ByteItem{
+			ByteItem: &dlppb.ByteContentItem{
+				Type: dlppb.ByteContentItem_TEXT_UTF8,
+				Data: []byte(textToInspect),
+			},
+		},
+	}
+
+	// Construct the custom regex detectors
+	var customInfoType = &dlppb.CustomInfoType{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_with_custom_regex.go:58]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_with_custom_regex.go [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+				customInfoType,
+			},
+			IncludeQuote: true,
+		},
+	}
+
+	// Send the request.
+	resp, err := client.InspectContent(ctx, req)
+	if err != nil {
+		fmt.Fprintf(w, "Receive: %v", err)
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_with_custom_regex.go:87]: issue: just return the error as-is to the caller.

**Review by telpirion — CHANGES_REQUESTED** [review]

File: dlp/snippets/inspect/inspect_test.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_test.go]: fixed it.

File: dlp/snippets/inspect/inspect_with_custom_regex.go (hunk 3 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_with_custom_regex.go]: fixed it.

File: dlp/snippets/inspect/inspect_with_custom_regex.go (hunk 4 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_with_custom_regex.go]: fixed it.

File: dlp/snippets/inspect/inspect_with_custom_regex.go (hunk 5 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_with_custom_regex.go]: fixed it.

File: dlp/snippets/inspect/inspect_test.go (hunk 1 above)
Comment (telpirion) [reply dlp/snippets/inspect/inspect_test.go:309]: Nope, it doesn't require the `new` keyword. `var` is fine. Please fix.

**Review by annadata — APPROVED** [review]
LGTM.

File: dlp/snippets/inspect/inspect_test.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_test.go]: Yes, It doesn't require `new` keyword, So Fixed it Here -> https://github.com/GoogleCloudPlatform/golang-samples/blob/a2ad801c6209726d71b3d1fa670936e0fb7fdd30/dlp/snippets/inspect/inspect_test.go#L362

**Review by telpirion — APPROVED** [review]

**Pull request merged**